plugins/package-managers/composer/src/main/kotlin/Composer.kt

```diff
@@ -156,7 +156,9 @@ class Composer(
         val virtualPackages = parseVirtualPackageNames(packages, projectPackageInfo, lockfile)
 
         val scopes = ALL_SCOPE_NAMES.mapTo(mutableSetOf()) { scopeName ->
-            parseScope(scopeName, projectPackageInfo, lockfile, packages, virtualPackages)
+            val requiredPackages = projectPackageInfo.getScopeDependencies(scopeName)
```

Comment:
Commit message: Please provide a rationale why the inlining makes sense. For example something like

> The function is only used once, and its parameter list is almost as long as the code itself.